include/rcpputils/thread_safety_annotations.h

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// NOTE: This is not the final resting place of this file - here to keep proof
+// of concept localized to a single package
+
+#ifndef RCPPUTILS__THREAD_SAFETY_ANNOTATIONS_H_
+#define RCPPUTILS__THREAD_SAFETY_ANNOTATIONS_H_
+
+#include <mutex>
+
+// Enable thread safety attributes only with clang.
+// The attributes can be safely erased when compiling with other compilers.
+// Prefixing all macros with R2 to avoid potential conflict with other projects
```

Comment:
They all should use the package name now that we have one. It's more verbose but much clearer for searchability and readability.

Reply:
Will do